### PR TITLE
Mhv-41714 creating filter box

### PR DESCRIPTION
**Pull request by robertroddev**
Search is being changed to filter, with that, the look and options have all been changed to reflect a filter flow rather than a search flow.

## Summary
- changed the mention of search to filter across the search components, and created a new component called FilterBox that handles the new filter functionality.
- changed the results message from "Displaying 10 of 100 conversations" to instead read "Displaying 10 of 100 messages" since the filter flow will not be displaying "conversations" instead it will be displaying messages
-  

## Related issue(s)
- [SM to VA.gov: SHOULD - Search box is changing to Filter box and other redesign work](https://vajira.max.gov/browse/MHV-41714)

## Testing done

- *unit test

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

| | Before | After |
| --- | --- | --- |
| Desktop |image|image|

## What areas of the site does it impact?
secure messaging

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/MessageList/MessageList.jsx [hunk 1]
```diff
@@ -194,7 +194,7 @@ const MessageList = props => {
       <div className="vads-u-padding-y--1 vads-l-row vads-u-margin-top--2 vads-u-border-top--1px vads-u-border-bottom--1px vads-u-border-color--gray-light">
         Displaying {displayNums[0]}
         &#8211;
-        {displayNums[1]} of {totalEntries} conversations
+        {displayNums[1]} of {totalEntries} messages
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/components/MessageList/MessageList.jsx:197]: You might want to check with UCD on this. It's not imperative to change now but in the phase 0 mocks changed the verbiage to 'conversations'.  If you decide to update it you may need to update some cypress & unit tests to pass.

File: src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx [hunk 2]
```diff
@@ -80,78 +80,86 @@ const SearchForm = props => {
...
+        <div className="keyword-help-text">{filterLabelBody}</div>
+        {!advancedOpen && (
+          <>
+            {searchTermError && (
+              <div className="error-message" role="alert">
+                <span className="sr-only">Error</span>
+                {searchTermError}
+              </div>
+            )}
+            {/* {filterLabelHeading + filterLabelBody} */}
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx:122]: I'd lose this comment

File: src/applications/mhv/secure-messaging/sass/search.scss [hunk 3]
```diff
@@ -1,20 +1,95 @@
...
     margin: 0 0 6px;
     @media (min-width: $medium-screen) {
       margin: 0 0 24px;
     }
   }
+  .filter-box{
+    margin-top: -30px;
+    margin-bottom: -20px;
+  }
+
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/sass/search.scss:85]: I saw that the input field was off center from the button.  
image
Try adding some css to strictly align the input with the button

File: src/applications/mhv/secure-messaging/util/actionTypes.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ export const Actions = {
     CREATE: 'SM_FOLDER_CREATE',
     DELETE: 'SM_FOLDER_DELETE',
     RENAME: 'SM_FOLDER_RENAME',
+    SET_FOLDER_ID: 'SM_FOLDER_SET_ACTIVE_FOLDER_ID',
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/util/actionTypes.js:31]: Maybe change this to 'GET_'.. it seems like you have to GET the Folder_ID in order to set it.

File: src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx [hunk 5]
```diff
@@ -80,78 +80,86 @@ const SearchForm = props => {
...
-            ? 'keyword-search-label'
-            : 'search-in-description'
+          resultsCount === undefined ? null : 'filter-results-in-folder'
         }
       >
-        {labelString}
+        {results}
       </label>
     );
   };
```
Comment (vsaleem) [inline src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx:103]: This is great!

File: src/applications/mhv/secure-messaging/components/MessageList/MessageList.jsx (hunk 1 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/components/MessageList/MessageList.jsx:197]: Hey yeah, i checked with Joel and Oleksii last week on this change, however I was not able to see the cypress test affected by this, would you happen to know any?

File: src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx (hunk 2 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx]: done 👍

File: src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+          modalTitle="Invalid search"
+          onPrimaryButtonClick={() => setFormError()}
+          onCloseEvent={() => setFormError()}
+          primaryButtonText="Ok"
+          status="error"
+          visible
+        >
+          <p>
+            Please use at least one of the following search fields or choose a
+            date range other than 'any'.
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx:133]: Not sure if we want to continue displaying errors in modals as we a re switching to inline errors across the board, let's raise this with UCD during validation

File: src/applications/mhv/secure-messaging/sass/search.scss (hunk 3 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/sass/search.scss:85]: Hey im not able to reproduce this. Are you using the review instance?

File: src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx [hunk 7]
```diff
@@ -109,6 +113,15 @@ const FolderThreadListView = props => {
     }
   }, 60000);
 
+  useEffect(
+    () => {
+      if (folderId) {
+        dispatch(setActiveFolderId(folderId));
+      }
+    },
+    [folder],
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx:122]: You may want to use `folderId` as a dependency variable here

File: src/applications/mhv/secure-messaging/reducers/folders.js [hunk 8]
```diff
@@ -1,6 +1,10 @@
 import { Actions } from '../util/actionTypes';
 
 const initialState = {
+  /**
+   * The current in-focus folderId, i.e. the folderId of the folder being viewed by the user
+   */
+  folderId: undefined,
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/reducers/folders.js:7]: not convinced we need to create this as we already have this captured in a `folder` object in this same reducer. Or is this somehow different?

File: src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+import { runAdvancedSearch } from '../../actions/search';
+import { dateFormat } from '../../util/helpers';
+
+const FilterBox = props => {
+  const {
+    folders,
+    testingSubmit,
+    testingDateRange,
+    testingFromDate,
+    testingToDate,
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx:25]: Are these props used just for the unit testing?

File: src/applications/mhv/secure-messaging/util/actionTypes.js (hunk 4 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/util/actionTypes.js:31]: Hey Vic thanks I appreciate the feedback! however the function(setActiveFolderId) that calls this isn't getting the folder Id, rather the folder id is being passed through as a parameter and is being set in state after that, so throughout the flow of this redux call there is no getting of the folder id, instead the folder Id has to have already been got prior to calling this action. You can see an example in the code in FolderThreadListView.jsx line 119.

File: src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx (hunk 5 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/components/Search/SearchForm.jsx:103]: Thank you! 💯

File: src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+            date range other than 'any'.
+          </p>
+          <ul>
+            <li>Message ID</li>
+            <li>From</li>
+            <li>Subject</li>
+            <li>Category</li>
+          </ul>
+        </VaModal>
+      )}
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx:142]: I am not sure this modal is no relevant though, as we are not using these criteria in this filter now, do we?

File: src/applications/mhv/secure-messaging/reducers/folders.js (hunk 8 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/reducers/folders.js]: yes you are correct, I went ahead and removed it

File: src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx (hunk 9 above)
Comment (robertbylight) [reply src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx]: i went ahead and deleted these props from the filterbox component, and the test cases.

**Review by alex-morgun-1 — APPROVED** [review]

**Comment by oleksii-morgun-va** [comment]
This PR covers a suggestion from [Midpoint Review - Design Feedback](https://github.com/department-of-veterans-affairs/va.gov-team/issues/51753)

**Pull request closed without merging**